### PR TITLE
add more Ruby versions to test matrix

### PR DESCRIPTION
**Pull request by digitaltom**
## Description

Please describe your change and provide as much context as possible.

Fixes # (issue)

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [ ] I have verified that my code follows RMT's coding standards with `rubocop`.
- [ ] I have reviewed my own code and believe that it's ready for an external review.
- [ ] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/281423/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 7dc9314f3796735a13fd867dad004770d01a6e94 into 2b1a3993d4047af2139130afb3e35569d93c38c5)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/281487/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e28442f8c5381cd1de211efcea0f1166561946fe into dff4a17709ce1bbcd6ecf7b1ede9f498970ef6f5)

**Comment by felixsch** [comment]
@digitaltom This is a enhanced version of https://github.com/SUSE/rmt/pull/860 Can we close the other one then?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/282966/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 75548de1b0b49a51ad5fa966448b61d50f3c2ea9 into aadaed39a04447634bb8b82ece1d48f0a75cd825)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/282971/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d2c11b60fc97681bfececf52a2f52972a4305dc6 into 33036395d5dd470e58783d6736840ee7f1393b4f)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/284127/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d2c11b60fc97681bfececf52a2f52972a4305dc6 into 8c856aa82749ed050a95b91af57d9523cd5d40fc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/284168/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d2c11b60fc97681bfececf52a2f52972a4305dc6 into a5d3e4cd67324f60b32bba68a2d5494677771e76)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/287314/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images.
Failed fast in project rmt on step docker-compose-build-images.
Failed fast in project rmt on step docker-compose-build-images.
Failed fast in project rmt on step docker-compose-build-images.
Failed fast in project rmt on step docker-compose-build-images.
Failed fast in project rmt on step docker-compose-build-images. (Merged d2c11b60fc97681bfececf52a2f52972a4305dc6 into 28209a15ff19ced4a3e9d210ea90bb5b7c9fe777)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/287325/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d2c11b60fc97681bfececf52a2f52972a4305dc6 into 92e8241ac6a217d415d725f2ef53e71deecd7d2f)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/287331/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d2c11b60fc97681bfececf52a2f52972a4305dc6 into 0649bb52467437b5c629796627c251e9b992e1f0)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/288461/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 18d0a2757ebbd8b84a6b0bcc3257d28090cae5b3 into 7c2c54ff8baeeb7ccff39943d284f2d40a1514a3)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/288489/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 18d0a2757ebbd8b84a6b0bcc3257d28090cae5b3 into 4207147aa63440b1f8702483a55b38809cba8d30)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/288654/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 18d0a2757ebbd8b84a6b0bcc3257d28090cae5b3 into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

File: .github/workflows/integrations.yml [hunk 1]
```diff
@@ -40,37 +44,48 @@ jobs:
...
+      - name: Setup files
         run: |
           sudo mkdir -p /var/lib/rmt
           ruby -e "require 'yaml'; puts({'database_test'=>{'host' => '127.0.0.1', 'port' => 3306, 'username'=>'rmt','password'=>'rmt','database'=>'rmt_test','adapter'=>'mysql2','encoding'=>'utf8','timeout'=>5000,'pool'=>5}}.to_yaml)" > config/rmt.local.yml
+          gem env
+          bundler_version=$(bundler -v | cut -d ' ' -f 3)
+          echo "Ruby version: $(ruby -v)"
+          echo "Available bundler: $(gem list | grep bundler)"
+          # dropping default bundler setting
+          rm /opt/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/gems/2.7.0/specifications/default/bundler-*.gemspec
```
Comment (felixsch) [inline .github/workflows/integrations.yml:66]: What is this patch and why does it needs to get removed. Is `/opt/hostedtoolcache` the bundler work dir path?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/289615/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fb07a45ecf3a7f4e28a8f86f560d37f9bc0d22bf into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290380/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged a5496259bdcdc2be09a9cd002f220bcdc3df219e into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290381/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 512305fcf29a0efda20299374f6292ff241a377f into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/290382/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 479b251bcc3b540cfcb65c990136309e4f59ac53 into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290383/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 11a77971b308fb7bcb8526792148c3bfc01e301d into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290384/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 4558545f2559dd0ab8be9d12fbe4de4f1a9ca5cd into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290385/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 9c24f181e7b1288402a4fb0595d7425652111024 into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

File: .github/workflows/integrations.yml (hunk 1 above)
Comment (digitaltom) [reply .github/workflows/integrations.yml]: I tried to force remove  the default bundler version, but that code is dropped now from the branch.

**Comment by digitaltom** [comment]
Failure with Ruby 3 is expected, because the RMT codebase is not ready yet. But we can see it now and fix it next.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290386/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged b6e6f7ddd6c8c9e8e97c410e581c1d5934270c81 into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/290387/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0b2353603938d36fe2df63b189f79fd829499a51 into 722e62d55d02aeda587f3c2cb23bf7896119b8cc)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/291629/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0b2353603938d36fe2df63b189f79fd829499a51 into d6e9d2f239a402c309422805b752ca2808a19ff1)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/291993/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0b2353603938d36fe2df63b189f79fd829499a51 into c93a445dffe6ad1e79144ca72dddc53fa2aaf236)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/293304/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0b2353603938d36fe2df63b189f79fd829499a51 into f3dd1349677e600d9259cbe07953c39ad7af6773)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/294110/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0b2353603938d36fe2df63b189f79fd829499a51 into 945532430284270631b092267ad77a06a0f046a2)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/294114/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 0b2353603938d36fe2df63b189f79fd829499a51 into dbf3b13be0f607e03635b64e0091b4a7b3fbc944)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/294178/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged ac86740ad84d33235a000ec718130088d0b22f5b into dbf3b13be0f607e03635b64e0091b4a7b3fbc944)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/294499/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged ac86740ad84d33235a000ec718130088d0b22f5b into e54da1b4500fddaa684869a1204a9d425b11154d)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/298860/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged ac86740ad84d33235a000ec718130088d0b22f5b into ecf699d0d093ee80b60035422c1d773abed5adc3)

File: .github/workflows/integrations.yml [hunk 2]
```diff
@@ -39,38 +43,29 @@ jobs:
...
-        id: gem-cache
-        with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-gems-
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
 
-      - name: Pre-run inline scripts
+      - name: Setup files
```
Comment (ngetahun) [inline .github/workflows/integrations.yml:48]: If we are renaming it, better to call it `Setup database config file`

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/301439/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e41f74ebbeb8776e80edb83983274bb2c67ecd04 into ecf699d0d093ee80b60035422c1d773abed5adc3)

File: .github/workflows/integrations.yml [hunk 3]
```diff
@@ -39,38 +43,29 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby-version }}
-          bundler-cache: true
-          bundler: 1.17
-
-      - name: "Cache gems"
```
[the remark this answers is not in the document]
Comment (digitaltom) [reply .github/workflows/integrations.yml:45]: This should already be covered with `bundler-cache: true` above.

File: .github/workflows/integrations.yml (hunk 2 above)
Comment (digitaltom) [inline .github/workflows/integrations.yml:48]:
```suggestion
      - name: Setup database config file
```

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/301440/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e41f74ebbeb8776e80edb83983274bb2c67ecd04 into 093d44593717b3aa60430eb94a1971eb21894cb0)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/301441/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fd49a2f4520d54eb8c6d81a8e823c2199579ae73 into 4746477768155a9cd7f3f33656b65705835e9df3)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/301572/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged fd49a2f4520d54eb8c6d81a8e823c2199579ae73 into 451dace5bf13f776b732a2e1c1a1dc111e9b912c)

**Comment by digitaltom** [comment]
> Do we have a follow-up PR to fix issues with ruby3 test pipeline?

No, not yet. That's why I marked the Ruby3 pipeline as not required to pass. It's the first step to have Ruby3 in the test matrix.

**Review by ngetahun — APPROVED** [review]
LGTM!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/301575/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d8364e90b7673a2f37aecc1a8bf48da01c6e7ae1 into ec9c46b298bc36b51d3e0ce55d63ccb5062af50f)

**Pull request merged**